src/main/java/ee/tuleva/onboarding/config/Oauth2RestTemplateConfiguration.java

```diff
@@ -0,0 +1,26 @@
+package ee.tuleva.onboarding.config;
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.oauth2.client.OAuth2ClientContext;
+import org.springframework.security.oauth2.client.OAuth2RestOperations;
+import org.springframework.security.oauth2.client.OAuth2RestTemplate;
+import org.springframework.security.oauth2.client.token.grant.client.ClientCredentialsResourceDetails;
+import org.springframework.security.oauth2.config.annotation.web.configuration.EnableOAuth2Client;
+
+@Configuration
+@EnableOAuth2Client
+public class Oauth2RestTemplateConfiguration {
+
+    @Bean
+    @ConfigurationProperties("epis.service.client")
+    ClientCredentialsResourceDetails oauth2ClientCredentialsResourceDetails() {
+        return new ClientCredentialsResourceDetails();
+    }
+
+    @Bean
+    public OAuth2RestOperations oauth2RestTemplate(ClientCredentialsResourceDetails resource, OAuth2ClientContext context) {
```

Comment:
I'm wondering if `EnableOAuth2Client` and `OAuth2ClientContext` are actually needed here 🤔 Doesn't it work without them?

Reply:
Not sure, but it's taken from the official spring javadocs: https://docs.spring.io/spring-security/oauth/apidocs/org/springframework/security/oauth2/config/annotation/web/configuration/EnableOAuth2Client.html

Reply:
This is for Authorization Code Grant, which definitely needs oauth context.

Reply:
You're right, it does work without the `@EnableOAuth2Client` and `OAuth2ClientContext` here. 🙇